lib/spack/spack/architecture.py

```diff
@@ -487,6 +487,27 @@ def arch_from_dict(d):
     return arch
 
 
+def get_full_system_from_platform():
```

Comment:
@tgamblin - if that contains os name, version, and the processor arch that's sufficient, yes.